tests/src/JIT/HardwareIntrinsics/X86/Avx2/Add/UInt64.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics.X86;
+using System.Runtime.Intrinsics;
+
+namespace HardwareIntrinsicTest
+{
+    public sealed unsafe class Test
```

Comment:
I would prefer not to split operation tests into separate test for each operation - data type combination. This would really explode number of projects used to build and run the tests multiplying project files without significant gain in test content. From my checks building tests in coreclr consists from tiny actual code file build time and huge MSBuild overhead. It is even more pronounced during test run, where core logic test run is a minuscule portion of the whole test running time. 

Splitting larger tests into tiny ones will only exacerbate these problems.

Reply:
> I would prefer not to split operation tests into separate test for each operation - data type combination. This would really explode number of projects used to build and run the tests multiplying project files without significant gain in test content. From my checks building tests in coreclr consists from tiny actual code file build time and huge MSBuild overhead. It is even more pronounced during test run, where core logic test run is a minuscule portion of the whole test running time.

Ideally we would be using an actual unit test framework (such as xUnit or NUnit), there would be a project per ISA (rather than a project per method or per overload), a class per method or overload, and a Fact per scenario.

However, I've not seen many other tests in mscorlib (outside of perf tests) that support this.

It might be worth changing this however. @Jkotas, thoughts?

Reply:
Unit test frameworks work great for testing framework APIs where it does not really matter how you call the API and it is fine to abstract it away using XUnit theories, etc.

They do not work well for testing codegen where you need the code calling the API to look in a particular way. (Also, the codegen/runtime bugs are pain to debug with unit testing framework loaded in the process since it introduces a lot of noise.)

I think you may want to look at autogenerating the test sources for testing the intrinsics.